src/textual/widgets/_data_table.py

```diff
@@ -1322,6 +1324,40 @@ def add_rows(self, rows: Iterable[Iterable[CellType]]) -> list[RowKey]:
             row_keys.append(row_key)
         return row_keys
 
+    def remove_row(self, row_key: RowKey | str) -> None:
+        """Remove a row (identified by a key) from the DataTable.
+
+        Args:
+            row_key: The key identifying the row to remove.
+
+        Raises:
+            RowDoesNotExist: If the row key does not exist.
+        """
+        if row_key not in self._row_locations:
+            raise RowDoesNotExist(f"Row key {row_key!r} is not valid.")
+
+        self._require_update_dimensions = True
```

Comment:
Will you not need `check_idle` after this? There is no guarantee an `on_idle` will follow, if this method is called in the context of another widget.